modules/azure/networkinterface.go

```diff
@@ -0,0 +1,180 @@
+package azure
+
+import (
+	"context"
+	"reflect"
+
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2019-09-01/network"
+	"github.com/gruntwork-io/terratest/modules/testing"
+	"github.com/stretchr/testify/require"
+)
+
+// NetworkInterfaceExists indicates whether the specified Azure Network Interface exists.
+// This function would fail the test if there is an error.
+func NetworkInterfaceExists(t testing.TestingT, nicName string, resGroupName string, subscriptionID string) bool {
+	exists, err := NetworkInterfaceExistsE(t, nicName, resGroupName, subscriptionID)
+	require.NoError(t, err)
+	return exists
+}
+
+// NetworkInterfaceExistsE indicates whether the specified Azure Network Interface exists.
+func NetworkInterfaceExistsE(t testing.TestingT, nicName string, resGroupName string, subscriptionID string) (bool, error) {
+	// Get the Network Interface
+	_, err := GetNetworkInterfaceE(t, nicName, resGroupName, subscriptionID)
+	if err != nil {
+		return false, err
```

Comment:
Is there some way to check the specific error that is returned when the network interface doesn't exist, and return `nil` for the error in that case? Otherwise, you would never get `false` in `NetworkInterfaceExists` because it would fail the test with the non-nil `err` value.

Reply:
I saw other examples check the error for a specific return type and did the same using the contains check. The "ResourceNotFound" code works for all of the resources. Other errors returned are "ResourceGroupNotFound" etc so am able to use the same check for all the resources. 

If there is a better way to get the error code please let me know.